subsys/net/ip/ipv4.c

```diff
@@ -187,6 +195,10 @@ int net_ipv4_parse_hdr_options(struct net_pkt *pkt,
 			break;
 		}
 
+		if (opts_len < opt_len) {
```

Comment:
There's exactly the same check just above the `switch` block, and neither of verified variables appear to be modified in between. Do we really need both checks?

Reply:
That's a fair point. Seems redundant to me too. @rveerama1 @jukkar any comments?

Reply:
+1. Not that it's logically "the same", syntactically it's `(opt_len > opts_len)`, so superfluous conditions here and above add quite a confusion. Let's add exactly the missing check(s). And to me, it looks like it's "if (opt_len < 2)" immediately after opt_len was read.

OTOH, I don't see explicit check for "options length is more that actual data", which commit message promises.

Reply:
yep, removed redundant.